mmdet/datasets/utils.py

```diff
@@ -98,3 +102,40 @@ def get_loading_pipeline(pipeline):
         'The data pipeline in your config file must include ' \
         'loading image and annotations related pipeline.'
     return loading_pipeline_cfg
+
+
+class CompatibleCheckHook(Hook):
+
+    def _check_head(self, model, dataset):
+        """Check whether the `num_classes` in head matches the length of
+        `CLASSSES` in `dataset`.
+
+        Args:
+            model (obj:`MMDataParallel` | obj:`nn.Module`): One stage
+                or Two stage Detector
+            dataset (obj:`CustomDataset`): Detection dataset.
+        """
+
+        for name, module in model.named_modules():
+            if hasattr(module,
+                       'num_classes') and not isinstance(module, RPNHead):
+                assert module.num_classes == len(dataset.CLASSES), \
+                    (f'The `num_classes` ({model.num_classes}) in {name}'
+                     f'does not matches the length of `CLASSES`'
+                     f'{len(dataset.CLASSES)}) in {dataset}')
+
+    def before_train_epoch(self, runner):
```

Comment:
Simply checking it before run is enough.

Reply:
It is  in `before_train_epoch` or `before_val_epoch` where we set `dataloader` to `runner`,   Because we don't not know which `dataloader`  would be used in  `before_run`